### PR TITLE
Add component for generate event attributes and event batching

### PR DESCRIPTION
**Pull request by yux0**
Add component for generate event attributes and event batching

### CODE REVIEW

File: common/xdc/event_generator.go [hunk 1]
```diff
@@ -35,40 +41,61 @@ type (
...
 		AddExitVertex(...Vertex)
+		// RandomEntryVertex is a random entry point which can be access at any state of the generator
 		AddRandomEntryVertex(...Vertex)
+		// AddModel loads model into the generator
+		// AddModel can load multiple models and models will be joint if there is common vertexes
 		AddModel(Model)
+		// HasNextVertex determines if there is more vertex to generate
 		HasNextVertex() bool
+		// GetNextVertex generates next vertex batch
 		GetNextVertex() []Vertex
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: GetNextVertices

File: common/xdc/event_generator.go [hunk 2]
```diff
@@ -35,40 +41,61 @@ type (
...
+		// ListGeneratedVertex lists the pasted generated vertexes
 		ListGeneratedVertex() []Vertex
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: ListGeneratedVertices

File: common/xdc/event_generator.go [hunk 3]
```diff
@@ -35,40 +41,61 @@ type (
...
+		// The name of the vertex. Usually, this will be the Cadence event type
 		SetName(string)
 		GetName() string
 		Equals(Vertex) bool
+		// IsStrictOnNextVertex means if the vertex must be followed by its children
+		// When IsStrictOnNextVertex set to true, it means this event can only follow by its neighbors
 		SetIsStrictOnNextVertex(bool)
 		IsStrictOnNextVertex() bool
+		// MaxNextVertex means the max neighbors can branch out from this vertex
+		// MaxNextVertex means the max neighbors can branch out from this vertex
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: copy past err

File: common/xdc/event_generator.go [hunk 4]
```diff
@@ -239,7 +304,7 @@ func (g *EventGenerator) getNextPossibleVertex() Vertex {
 		}
 		neighbors := g.connections[leaf]
 		for _, nextV := range neighbors {
-			if nextV.GetCondition() == nil || (nextV.GetCondition() != nil && nextV.GetCondition()(g.previousVertexes)) {
+			if nextV.GetCondition() == nil || (nextV.GetCondition() != nil && nextV.GetCondition()()) {
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: this is equivalent to `nextV.GetCondition() == nil ||  nextV.GetCondition()()`

Comment (wxing1292) [reply common/xdc/event_generator.go]: on line 312, it it guaranteed to be `isAccessible == false` ?

File: common/xdc/event_generator.go [hunk 5]
```diff
@@ -239,7 +304,7 @@ func (g *EventGenerator) getNextPossibleVertex() Vertex {
...
 				isAccessible = true
 				break
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: once `isAccessible == true` this function can basically return?

File: common/xdc/event_generator.go [hunk 6]
```diff
@@ -26,6 +26,12 @@ import (
...
 
+var (
+	defaultBatchFunc = func(batch []Vertex) bool {
+		return len(batch) == 0
+	}
+)
+
 type (
 	// Model represents a state transition graph that contains all the relationships of Vertex
 	Model interface {
```
Comment (wxing1292) [inline common/xdc/event_generator.go]: it will make the code cleaner if different interfaces / implementations are put in different files

Comment (yux0) [reply common/xdc/event_generator.go]: Moved to a different file

File: common/xdc/event_generator.go (hunk 5 above)
Comment (yux0) [reply common/xdc/event_generator.go]: Yes. This function is to check if there is a path to move forward. I updated the break to a reurn

File: common/xdc/event_generator.go (hunk 4 above)
Comment (yux0) [reply common/xdc/event_generator.go]: On 312, yes it is guaranteed or it will break the loop

**Pull request merged**